### ISSUE TITLE
what happened to $resetChanged(function() { }); in 0.8.26

### ISSUE THREAD

**Issue by philjones88** [AngularAgility/AngularAgility#132]
Hi,

I can see in 0.8.26 you removed the ability to do:

`js
$aaFormExtensions.$resetChanged(function () {
 // do something
// e.g. we do $state.go('blah'); here, inside the function to stop changed warning confirms popping up
});
`

What's the reason behind this and what's the alternative now?

We have the scenario of: user edits form, user hits save, save method calls API, redirect.

To stop the "unsaved changes" dialog appearing we used the $resetChanged callback to then do the redirect.

**Comment by johnculviner** [comment]
Hey, the thought was it wasn't necessary anymore (fixed). Try calling whatever you were gonna call right after it. Let me know if that doesn't work. I thought w/o knowledge  of the API it would be alot more sensical to not require that callback if I could swing it

**Comment by philjones88** [comment]
Ok great, I'll alter all our calls to it and let you know. I'll admit my first attempt was to not use a callback and then I had to lookup that you did a callback. So a good change :)

**Comment by johnculviner** [comment]
Yeah it might not be fixed for all cases I've discovered but might be for yours. It was a stupid hack to begin with and I wanted to get rid of it :)

**Comment by philjones88** [comment]
A weird issue, between 0.8.27 and 0.8.28 the "unsaved changes" stuff seems to have broken. Using bower to swap between the two, on .27 the unsaved changes works and on .28 it doesn't. Anything else need updating?

**Comment by johnculviner** [comment]
Hmmm nothing really changed there.... ?

https://github.com/AngularAgility/AngularAgility/commit/fb6e9fa82cc2bee5275a7498f510f67360c59fa5

**Comment by philjones88** [comment]
I know. Which is very weird. I did look at the diff. I'll double check my findings tomorrow...

It just doesn't pop up and I don't see an error in the console.

**Comment by philjones88** [comment]
updating to 0.8.28 and removing the one change:

```js
|| config.cached
```

Works. With that change I don't get any unsaved changes warning dialogs.

Perhaps it's the way I'm using AA? But it's pretty standard, settings a custom labelStrategy and a custom fieldGroupStrategy in the NG .config section...

Guess I'll stick with 0.8.27 for now then :(

**Comment by philjones88** [comment]
Seems rolling back to 0.8.27 gives me unsaved changes alerts but also the ```resetChanged()``` doesn't clear the status, I'll try and make a replication project for this but it's just being called inside the success function of ```then``` inside a $http.

**Comment by johnculviner** [comment]
ahh been super busy, not sure why that could be causing it. I'll try to dig in here at some point. the reason why I put that in there was because loading was getting fired when it wasn't really loading (aka you had a template cached already). This resulted in a flash of aaIsLoading being true even though said item was already loaded!

**Comment by smajl** [comment]
Running 0.8.28 here and I just realized, that my app stopped asking about unsaved changes (when canceling form editing and navigating away). Why? Because `$aaFormExtensions.$changed` stays always `false` even after data changes. It's broken. :( I can confirm that rolling back to 0.8.27 fixes this! That's funny, looking at the 27/28 diff. :D

**Comment by johnculviner** [comment]
ahh thats very strange, yeah how the heck does that work! Let me do a reproduction here and at the very least i might back out that change because that is way more minor than this lib not working right for unsaved changes!

**Comment by SFGuru** [comment]
i am getting error after update angular 1.6.1 or AngularAgility(0.8.37)
The error on this line form.$aaFormExtensions.$clearErrors();

TypeError: Cannot read property '$$controls' of undefined
at $setUntouched (angular.js:22874)
    at Object.$clearErrors (angular-agility.js:2598)
    at Scope.$scope.editModeCancel (.js:77)
    at fn (eval at compile (angular.js:15156), <anonymous>:4:229)
    at callback (angular.js:26744)
    at Scope.$eval (angular.js:17972)
    at Scope.$apply (angular.js:18072)
    at HTMLButtonElement.<anonymous> (angular.js:26749)
    at HTMLButtonElement.dispatch (jquery.js:4737)
    at HTMLButtonElement.elemData.handle (jquery.js:4549)
